openfl-workspace/keras_cnn_mnist/src/tfmnist_inmemory.py

```diff
@@ -27,9 +30,13 @@ def __init__(self, data_path, batch_size, **kwargs):
         # Then we have a way to automatically shard based on rank and size of
         # collaborator list.
 
-        _, num_classes, X_train, y_train, X_valid, y_valid = load_mnist_shard(
-            shard_num=int(data_path), **kwargs
-        )
+        try:
+            _, num_classes, X_train, y_train, X_valid, y_valid = load_mnist_shard(
+                shard_num=int(data_path), **kwargs
+            )
+        except ValueError:
+            logger.error("Please pass the shard number (integer) for the collaborator using data path flag.")
+            return
```

Comment:
This block will raise `ValueError` on any downstream error that may occur within `try` block. Our intention is only to check the typecast.

Recommended way:
```python
try:
  shard_num = int(data_path)
except:
  raise ValueError(...)  # auto-exits.

_, num_classes, X_train, y_train, X_valid, y_valid = load_mnist_shard(
  shard_num=shard_num, **kwargs)
```